src/table/table.scss

```diff
@@ -99,6 +103,12 @@
     }
   }
 
+  .iui-reorder-column {
+    @include themed {
+      border-right: $iui-xxs solid t(iui-color-foreground-primary);
```

Comment:
Not sure if its always right though

https://github.com/iTwin/iTwinUI-react/pull/561/files#r814875116

Reply:
Updated to work on left or right.